muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/chat/ChatClientHolder.java

```diff
@@ -0,0 +1,69 @@
+package fi.otavanopisto.muikku.plugins.chat;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.ConcurrencyManagement;
+import javax.ejb.ConcurrencyManagementType;
+import javax.ejb.Schedule;
+import javax.ejb.Singleton;
+import javax.inject.Inject;
+
+import rocks.xmpp.core.session.XmppClient;
+
+@Singleton
+@ConcurrencyManagement(ConcurrencyManagementType.BEAN)
```

Comment:
Is this necessary, iirc the default is write lock anyways so no public method can be called at the same time.